xds/internal/xdsclient/tests/cds_watchers_test.go

```diff
@@ -41,6 +41,26 @@ import (
 	v3discoverypb "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v3"
 )
 
+type clusterWatcher struct {
+	updateCh *testutils.Channel
+}
+
+func newClusterWatcher() *clusterWatcher {
+	return &clusterWatcher{updateCh: testutils.NewChannel()}
+}
+
+func (ew *clusterWatcher) OnUpdate(update *xdsresource.ClusterResourceData) {
+	ew.updateCh.Send(xdsresource.ClusterUpdateErrTuple{Update: update.Resource})
+}
+
+func (ew *clusterWatcher) OnError(err error) {
+	ew.updateCh.SendOrFail(xdsresource.ClusterUpdateErrTuple{Err: err})
```

Comment:
Why does this SendOrFail where as API you switched from Sends on all? Can you switch this to blocking and assert it doesn't receive an error from updateCh? I don't really care, but can you comment here or top level for clusterWatcher if this route as to why one send blocks and one doesnt?

Reply:
I actually wanted to use `Replace`, which is what I had used in the xds_resolver_test. This block of code, I just copied over from the `endpointsWatcher`. Changed both to use `Replace` instead and added a comment as well. Thanks.